stargz/verify/verify.go

```diff
@@ -0,0 +1,225 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package verify
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strconv"
+
+	"github.com/google/crfs/stargz"
+	"github.com/pkg/errors"
+)
+
+const (
+	// TOCJSONDigestAnnotation is an annotation for image manifest. This stores the
+	// digest of the TOC JSON
+	TOCJSONDigestAnnotation = "containerd.io/snapshot/stargz/toc.digest"
+)
+
+// jtoc is a TOC JSON schema which contains additional fields for chunk-level
+// content verification. This is still backward compatible to the official
+// definition:
+// https://github.com/google/crfs/blob/71d77da419c90be7b05d12e59945ac7a8c94a543/stargz/stargz.go
+type jtoc struct {
+	Version int         `json:"version"`
```

Comment:
Needs documentation

Reply:
Seems unused field?

Reply:
Added documentation. `Version` is unused in this package but we need it for keeping the backwards-compatibility with stargz.